providers/implementations/kem/ml_kem.c

```diff
@@ -15,218 +15,217 @@
 #include <openssl/params.h>
 #include <openssl/err.h>
 #include <openssl/proverr.h>
+#include "crypto/ml_kem.h"
 #include "prov/provider_ctx.h"
 #include "prov/implementations.h"
 #include "prov/securitycheck.h"
 #include "prov/providercommon.h"
-#include "prov/mlkem.h"
 
-#define BUFSIZE 1000
-#if defined(NDEBUG) || defined(OPENSSL_NO_STDIO)
-/* TODO(ML-KEM) to remove or replace with TRACE */
-static void debug_print(char *fmt, ...)
-{
-}
-#else
-static void debug_print(char *fmt, ...)
-{
-    char out[BUFSIZE];
-    va_list argptr;
-
-    va_start(argptr, fmt);
-    vsnprintf(out, BUFSIZE, fmt, argptr);
-    va_end(argptr);
-    if (getenv("TEMPLATEKM"))
-        fprintf(stderr, "TEMPLATE_KM: %s", out);
-}
-#endif
+static OSSL_FUNC_kem_newctx_fn ml_kem_newctx;
+static OSSL_FUNC_kem_freectx_fn ml_kem_freectx;
+static OSSL_FUNC_kem_encapsulate_init_fn ml_kem_encapsulate_init;
+static OSSL_FUNC_kem_encapsulate_fn ml_kem_encapsulate;
+static OSSL_FUNC_kem_decapsulate_init_fn ml_kem_decapsulate_init;
+static OSSL_FUNC_kem_decapsulate_fn ml_kem_decapsulate;
+static OSSL_FUNC_kem_set_ctx_params_fn ml_kem_set_ctx_params;
```

Comment:
The settables function `ml_kem_settable_ctx_params` is missing here.

Reply:
Added.